### PR TITLE
Update JavaSparkContextVarargsWorkaround.java 

### PR DESCRIPTION
**Pull request by SakalyaDeshpande**
Taking common functionality into a Single method to avoid duplication of code.

### CODE REVIEW

File: core/src/main/java/org/apache/spark/api/java/JavaSparkContextVarargsWorkaround.java [hunk 1]
```diff
@@ -58,6 +52,14 @@ public JavaDoubleRDD union(JavaDoubleRDD... rdds) {
     }
     return union(rdds[0], rest);
   }
+  
+  public final List<T> populateRDDList(T rdds) {
```
Comment (srowen) [inline core/src/main/java/org/apache/spark/api/java/JavaSparkContextVarargsWorkaround.java]: OK, pretty trivial but a win. You can use this in line 49 too right?

File: core/src/main/java/org/apache/spark/api/java/JavaSparkContextVarargsWorkaround.java [hunk 2]
```diff
@@ -52,11 +46,16 @@ public JavaDoubleRDD union(JavaDoubleRDD... rdds) {
     if (rdds.length == 0) {
       throw new IllegalArgumentException("Union called on empty list");
     }
-    List<JavaPairRDD<K, V>> rest = new ArrayList<>(rdds.length - 1);
+    rest = populateRDDList(rdds);
+    return union(rdds[0], rest);
+  }
+  
+  public static final List<T> populateRDDList(T rdds) {
```
Comment (srowen) [inline core/src/main/java/org/apache/spark/api/java/JavaSparkContextVarargsWorkaround.java]: One last thing -- `private`?  And doesn't this have to be `T... rdds` to work?

Comment (sakky11) [inline core/src/main/java/org/apache/spark/api/java/JavaSparkContextVarargsWorkaround.java]: I agree with making the method "private".But rdds is not any varargs so why do we need ...??

Comment (srowen) [inline core/src/main/java/org/apache/spark/api/java/JavaSparkContextVarargsWorkaround.java]: This doesn't build. You're using `T` like an array type but it isn't. It has to be an array type as far as the method is concerned. `T[]` would be fine as the type.

Comment (sakky11) [inline core/src/main/java/org/apache/spark/api/java/JavaSparkContextVarargsWorkaround.java]: I created a build with T...

Comment (sakky11) [inline core/src/main/java/org/apache/spark/api/java/JavaSparkContextVarargsWorkaround.java]: will this work ??

**Comment by sakky11** [comment]
Any update on the recent change ??

**Comment by srowen** [comment]
Jenkins test this please

**Comment by srowen** [comment]
Yeah, still doesn't compile. Are you compiling this locally?

**Comment by sakky11** [comment]
Added the required fix.Can you please start the build ??

**Comment by srowen** [comment]
@sakky11 you have not even addressed the compilation error. This is wasting everyone's time. Please _compile_ this locally and fix it before pinging again.

**Comment by sakky11** [comment]
Hey sorry @srowen ,Actually I have fixed the compilation error but missed it in adding in a pull request.Sorry.

**Comment by sakky11** [comment]
I think the issue should be fixed now.Sorry for trouble.

**Comment by srowen** [comment]
Did you compile this locally before pushing?

**Comment by JoshRosen** [comment]
This patch is only a one-line reduction of code and touches a file which hasn't needed to be modified in ages; is this change really necessary / worth spending review time on?

**Comment by sakky11** [comment]
I know its trivial but it still avoids duplication of code and make it more readable.

**Comment by sakky11** [comment]
Any update on this commit ??

**Comment by srowen** [comment]
@sakky11 I asked you because I can already see that your change doesn't even compile, for a third time. This is a trivial change, and not worth spending more time discussing. Please close this, and read https://cwiki.apache.org/confluence/display/SPARK/Contributing+to+Spark before opening another change. At minimum, you need to make sure your change even builds before considering opening a pull request.

**Pull request closed without merging**